Origin: EmberBSD, AI-assisted.
sendmail's db2 and db4 options used databases/db and databases/db4;
EmberBSD keeps one Berkeley DB, db18. One bdb option (legacy db2 and
db4 names map to it) builds the NEWDB maps against db18. Not submitted
upstream.

--- a/mail/sendmail/Makefile
+++ b/mail/sendmail/Makefile
@@ -90,11 +90,8 @@ post-patch: make-sendmail-siteconfig
 .if !empty(PKG_OPTIONS:Mldap)
 	cat ${FILESDIR}/site.config.m4-ldap >>${SITECONFIG}
 .endif
-.if !empty(PKG_OPTIONS:Mdb2)
-	cat ${FILESDIR}/site.config.m4-db2 >>${SITECONFIG}
-.endif
-.if !empty(PKG_OPTIONS:Mdb4)
-	cat ${FILESDIR}/site.config.m4-db4 >>${SITECONFIG}
+.if !empty(PKG_OPTIONS:Mbdb)
+	cat ${FILESDIR}/site.config.m4-bdb >>${SITECONFIG}
 .endif
 .if !empty(PKG_OPTIONS:Mtls)
 	cat ${FILESDIR}/site.config.m4-starttls >>${SITECONFIG}
--- /dev/null
+++ b/mail/sendmail/files/site.config.m4-bdb
@@ -0,0 +1,3 @@
+# enable Berkeley DB (db18)
+APPENDDEF(`confENVDEF', `-DNEWDB -I${BUILDLINK_DIR}/include/db18')
+APPENDDEF(`confLIBS', `-ldb18')
--- a/mail/sendmail/files/site.config.m4-db2
+++ /dev/null
@@ -1,5 +0,0 @@
-# $NetBSD: site.config.m4-db2,v 1.5 2004/08/30 20:16:28 adrianp Exp $
-
-# enable db2
-APPENDDEF(`confENVDEF', `-DNEWDB -I${BUILDLINK_DIR}/include/db2')
-APPENDDEF(`confLIBS', `-ldb2')
--- a/mail/sendmail/files/site.config.m4-db4
+++ /dev/null
@@ -1,5 +0,0 @@
-# $NetBSD: site.config.m4-db4,v 1.2 2004/08/30 20:16:28 adrianp Exp $
-
-# enable db4
-APPENDDEF(`confENVDEF', `-DNEWDB -I${BUILDLINK_DIR}/include/db4')
-APPENDDEF(`confLIBS', `-ldb4')
--- a/mail/sendmail/options.mk
+++ b/mail/sendmail/options.mk
@@ -1,7 +1,7 @@
 # $NetBSD: options.mk,v 1.28 2026/03/02 07:39:29 jnemeth Exp $
 
 PKG_OPTIONS_VAR=	PKG_OPTIONS.sendmail
-PKG_SUPPORTED_OPTIONS=	inet6 db2 db4 ldap sasl tls tcpwrappers
+PKG_SUPPORTED_OPTIONS=	inet6 bdb ldap sasl tls tcpwrappers
 PKG_SUPPORTED_OPTIONS+=	sendmail-ffr-tls sendmail-ffr-badrcptshutdown
 PKG_SUPPORTED_OPTIONS+=	smtputf8
 PKG_SUGGESTED_OPTIONS=	inet6 tcpwrappers tls
@@ -18,16 +18,16 @@ PKG_OPTIONS_LEGACY_OPTS+=	blacklist:blocklist
 .endfor
 
 PKG_OPTIONS_LEGACY_OPTS+=	starttls:tls ffr_tls_1:sendmail-ffr-tls
+PKG_OPTIONS_LEGACY_OPTS+=	db2:bdb db4:bdb
 
 .include "../../mk/bsd.options.mk"
 
 ###
-### Berkeley DB version 2/4 format for on disk databases e.g. aliases
+### Berkeley DB format for on disk databases e.g. aliases
 ###
-.if !empty(PKG_OPTIONS:Mdb2)
-.  include "../../databases/db/buildlink3.mk"
-.elif !empty(PKG_OPTIONS:Mdb4)
-.  include "../../databases/db4/buildlink3.mk"
+.if !empty(PKG_OPTIONS:Mbdb)
+BDB_ACCEPTED=	db18
+.  include "../../mk/bdb.buildlink3.mk"
 .endif
 
 ###
